libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1195,6 +1195,12 @@ export class BotFrameworkAdapter
             status = 400;
             const request = await parseRequest(req);
 
+            if (!req.body.type) {
+                console.warn('BadRequest: Missing activity or activity type.');
+                status = StatusCodes.BAD_REQUEST;
+                return;
+            }
+
```

Comment:
I'm not sure of adding this here. In case of an error in parseRequest, would this code be reached?